language/stdlib/modules/libra_account.move

```diff
@@ -631,6 +633,68 @@ module LibraAccount {
         make_account<Token, Empty::T>(new_account, auth_key_prefix, Empty::create(), false)
     }
 
+
+    ///////////////////////////////////////////////////////////////////////////
+    // Designated Dealer API
+    ///////////////////////////////////////////////////////////////////////////
+
+    public fun is_designated_dealer(addr: address): bool acquires Role {
+        let dealer =
+            &mut borrow_global_mut<Role<DesignatedDealer::Dealer>>(addr).role_data;
+        DesignatedDealer::is_designated_dealer(dealer)
+    }
+
+    public fun remove_designated_dealer(addr: address) acquires Role {
+        let dealer =
+            &mut borrow_global_mut<Role<DesignatedDealer::Dealer>>(addr).role_data;
+        DesignatedDealer::remove_designated_dealer(dealer)
+    }
+
+    public fun add_tier(addr: address, tier_upperbound: u64) acquires Role {
+        let dealer =
+            &mut borrow_global_mut<Role<DesignatedDealer::Dealer>>(addr).role_data;
+        DesignatedDealer::add_tier(dealer, tier_upperbound)
+    }
+
+    public fun update_tier(addr: address, tier_index: u64, new_upperbound: u64) acquires Role {
+        let dealer =
+            &mut borrow_global_mut<Role<DesignatedDealer::Dealer>>(addr).role_data;
+        DesignatedDealer::update_tier(dealer, tier_index, new_upperbound)
+    }
```

Comment:
I might be off here, but why do we want these functions in LibraAccount?

Reply:
This is my fault: an artifact of the way that roles are currently implemented (every `LibraAccount::Role<T>` resource with a `T` for each role). @valerini and I were brainstorming a design pattern to implement this that won't require this pesky wrappers, but probably best to move forward with this approach for now.

Reply:
The role resource (maintaining Dealer state) is only accessible from within the LibraAccount for now - not just that wrappers are needed but this file will blow up if every function needing account state needs to originate here :)